### DISCUSSION TITLE
How to generate composite beam?

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#23501] [Q&A General]
Hello MOOSE experts,

I am trying to solve beam equation

https://mooseframework.inl.gov/source/kernels/StressDivergenceBeam.html

I can able to solve it as single layer/material. Is it possible to add second layer of some other material for this module?

For example,
I want to simulate something like bimetallic thermal strip with eigen strain given to the upper layer.

Any suggestions?

Best,
Avtar

**Comment by GiudGiud** [comment]
Hello

Did you get a working solution here?

Guillaume

**Reply by avtarsinghh1991** [reply]
@GiudGiud I did solve it but analytically with lots of assumptions. Still looking if I can do it using MOOSE.

Thanks,
Avtar

**Reply by GiudGiud** [reply]
More than likely so.
Does it work with the same assumptions as analytically?
